code/datums/elements/immerse.dm

```diff
@@ -0,0 +1,387 @@
+/**
+ * A visual element that makes movables entering the attached turfs look immersed into that turf.
+ *
+ * Abandon all hope, ye who read forth, for this immerse works on mind-numbing workarounds,
+ */
+/datum/element/immerse
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY | ELEMENT_BESPOKE
+	argument_hash_start_idx = 2
+	///An association list of turfs that have this element attached and their affected contents.
+	var/list/attached_turfs_and_movables = list()
```

Comment:
this really smells like something that would be better done as a component, do you expect a high volume here?

Reply:
Unless the beach away mission is loaded, the number of water turfs that gets loaded into the game world is fairly low actually. However I've left this as an element since it can be attached to turfs (let's hypothesize I want this attached to lava one day when I'm 1000% it's safe it would be a good idea) and that having this a component adds an additional step of either using a signal or GetComponent() just to move tracked content from one list to another.